### PR TITLE
Awu/threejs example

### PR DESCRIPTION
**Pull request by asunawu**
image

### CODE REVIEW

File: integrations/threejs/threejs.html [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+          var v = camera.projectionMatrix.elements;
+          udSDKJS_SetMatrix("projection", v[0],v[1],v[2],v[3],v[4],v[5],v[6],v[7],v[8],v[9],v[10],v[11],v[12],v[13],v[14],v[15]);
+
+          udSDKJS_RenderQueue();
+          var ptr = udSDKJS_GetColourBuffer();
+          var data = new Uint8Array(
+            HEAPU8.subarray(ptr, ptr + width * height * 4)
+          );
+
+          colourTextureGL = new THREE.DataTexture(
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:158]: Is it possible to create this (and the depth texture below) once (on resize), and then upload the pixels every frame?

File: integrations/threejs/threejs.html [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+      void main(void) {
+        // Correction work
+        //topLeft = texture(u_texture, vec2(0.0, 0.0));
+        //// We want to check if the magenta point is rendered in top left corner
+        //// If not, flip it
+        //if (topLeft == magenta)
+        //  v_uv = vec2(aVertexUVs.x, aVertexUVs.y);
+        //else
+        //  v_uv = vec2(aVertexUVs.x, 1.0 - aVertexUVs.y);
+		v_uv = position.xy * vec2(0.5, 0.5) + vec2(0.5, 0.5);
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:31]: Spacing.

File: integrations/threejs/threejs.html [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+      // A magenta color for referencing
+      // Refer to line 311 to 315, we set a point to magenta color.
+      // And we will compare the magenta point with this vec4 magenta.
+      vec4 magenta = vec4(1.0, 0.0, 1.0, 1.0);
+      vec4 topLeft;
+
+      void main(void) {
+        // Correction work
+        //topLeft = texture(u_texture, vec2(0.0, 0.0));
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:24]: Remove commented code.

File: integrations/threejs/threejs.html [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <title>My first three.js app</title>
+    <style>
+      body {
+        margin: 0;
+      }
+    </style>
+    <script id="shader-vs" type="x-shader/x-vertex">
+      //uniform sampler2D u_texture;
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:12]: Remove commented code.

File: integrations/threejs/threejs.html [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+      precision mediump float;
+      precision highp int;
+      
+      //Input Format
+      in highp vec2 v_uv;
+
+      uniform sampler2D u_texture;
+      uniform sampler2D u_depth;
+      
+      //layout(location = 0) out highp vec4 out_var_SV_Target0;
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:45]: Remove commented code.

File: integrations/threejs/threejs.html [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+      const vertices = new Float32Array([-1,-1,0,  1,-1,0,  1,1,0, -1,-1,0,  1,1,0,  -1,1,0]);
+      geometry.setAttribute(
+        "position",
+        new THREE.BufferAttribute(vertices, 3)
+      );
+      mesh = new THREE.Mesh(geometry, material);
+      mesh.position.set( -0.5, -0.5, 0 );
+      scene.add(mesh);
+
+      ///// test
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:121]: More meaningful comment please.

E.g. "Add a simple texture box to test intersection with Unlimited Detail".

File: integrations/threejs/threejs.html [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+      const boxGeometry = new THREE.BoxGeometry( 0.25,  0.25,  0.25 );
+      const texture = new THREE.TextureLoader().load( 'crate.gif' );
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:123]: I would change the box to untextured. Then you don't need to add 'crate.gif' into the sample.

File: integrations/threejs/threejs.html [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+            dataHeapF32,
+            width,
+            height,
+          );
+          udTextureDepth.type = THREE.FloatType;
+          udTextureDepth.format = THREE.RedFormat;
+          udTextureDepth.needsUpdate = true;
+
+          material.uniforms.u_texture.value = colourTextureGL;
+          material.uniforms.u_depth.value = udTextureDepth; // however all data are 1 in this array, not sure why
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:182]: Is this comment still relevant?

File: integrations/threejs/threejs.html [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+
+          ///test
```
Comment (peterAdams1014) [inline integrations/threejs/threejs.html:184]: Better comment please.

**Review by peterAdams1014 — CHANGES_REQUESTED** [review]

File: integrations/threejs/threejs.html (hunk 6 above)
Comment (asunawu) [reply integrations/threejs/threejs.html]: image

File: integrations/threejs/threejs.html (hunk 1 above)
Comment (asunawu) [reply integrations/threejs/threejs.html]: fixed.

**Review by peterAdams1014 — APPROVED** [review]

**Pull request merged**